src/abio/io.cljs

```diff
@@ -195,11 +204,33 @@
   "A tree seq on files"
   [dir]
   (tree-seq
-    (fn [f] (-directory? *io-bindings* (:path f)))
-    (fn [d] (map as-file
-              (-list-files *io-bindings* (:path d))))
-    (as-file dir)))
-
+   (fn [f] (-directory? *io-bindings* (:path f)))
+   (fn [d] (map as-file
+                (-list-files *io-bindings* (:path d))))
+   (as-file dir)))
+
+;; XXX Passing this a string path and opts fails, but passing an existing
+;;  BufferedReader works.
+;;
+;; (defn waittest [ms]
+;;   (let [br (io/reader testpath :encoding "UTF8") ; you have to define testpath first
+;;         wait-fn #(do (println (io/-read-line br))(io/-close br))]
+;;     (js/setTimeout wait-fn ms)))
+;;
+;; If you wait 3 milliseconds, then this works properly. Since `readable.read`
+;; -- see https://nodejs.org/api/stream.html#stream_readable_read_size --
+;; will return `null`/`nil` if the buffer is empty when you call `read`, I'm
+;; assuming this has to do with the time it takes to set up the stream and get
+;; everything primed.
+;;
+;; This test was done with Node `ReadableStream`s. If we want to use streams in
+;; Lumo/Node, we need to attach to the events emitted by the streams. As such,
+;; it's not feasible to make synchronous streams (though maybe with
+;; async/await...) on a Node host.
```

Comment:
Don't fully understand this comment.. I think `async/await` is basically a wrapper to promises and you would not get the streaming behavior...but of course, there is probably something I am missing :)

Reply:
I see, you kind of would like to transform a stream to a sync call? I was never able to achieve that myself...

Reply:
To be honest I'm not sure async/await would be feasible. Dealing with streams in Node, from what I understand, involves registering callbacks to particular events, and if the goal would be some straight line creation then reading from a stream, having callbacks throws all of that out the window.

That being said, I've started thinking of ways to tie the callbacks and the reading functions together in streams using record-local atoms and loops, though that's a different PR all-together :)